### ISSUE TITLE
JSX cannot use tag names that start with "public-"

### ISSUE THREAD

**Issue by lyonbot** [microsoft/TypeScript#46512]
# Bug Report

### 🔎 Search Terms

JSX public private keywords reserved strict tag tagname

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about JSX

### ⏯ Playground Link

https://www.typescriptlang.org/play?ts=4.4.4#code/CYUwxgNghgTiAEYD2A7AzgF3gJRFMGAXPFCgJ4BQoksCKUAtiGgA74IBSAygBrwDeFePACWKDCBgAzdvACS4mGLQiwAUQggm4tAKHD4AbQDWxTEpQBzALrFSlYQF8KzipqxR4AXngAeYCIAbgB8+r4sAK4ARhCqALQA7nFgsMDBvgD0kTHxSSkwaRSZASFAA

### 💻 Code

```ts
declare const React: any
declare namespace JSX {
  interface IntrinsicElements {
    [k: string]: any
  }
}

// ------------------------------
// below is the code

let a = <div>
  <public-card></public-card>
</div>
```

### 🙁 Actual behavior

It correctly compiled to `React.createElement("div", null, React.createElement("public-w-card", null))`.

However, `public` of the tag is highlighted and errors are reported:

- Identifier expected. 'public-card' is a reserved word in strict mode.

### 🙂 Expected behavior

Code compiled with no error.
